aten/src/ATen/native/TensorFactories.h

```diff
@@ -101,7 +101,14 @@ inline void check_supported_max_int_with_precision(int64_t n, const Tensor& tens
 // fill the tensor with NaN if it is floating point or complex type, or fill
 // with max value if it is integer type
 inline Tensor& fill_empty_deterministic_(Tensor& tensor) {
-  if (tensor.is_floating_point() || tensor.is_complex()) {
+  if (tensor.scalar_type() == ScalarType::Float8_e5m2 ||
+      tensor.scalar_type() == ScalarType::Float8_e4m3fn) {
+    AT_DISPATCH_FP8_TYPES(
+      tensor.scalar_type(), "fill_empty_deterministic_", [&]() {
```

Comment:
does fill_empty_deterministic not work  for float8? 
cc @malfet

Reply:
An issue is "tensor.is_floating_point() || tensor.is_complex()" is True for fp8 but then AT_DISPATCH_FLOATING_AND_COMPLEX_TYPES_AND2 don't handle fp8 so raises an error. And not sure if fp8 has quite_NaN.